### ISSUE TITLE
Error in decryption result after decimal integer secret sharing

### ISSUE THREAD

**Issue by ConRresIm** [tf-encrypted/tf-encrypted#824]
I use the pond protocol for secret sharing, but when I decrypt, the result is wrong。The error mainly occurs when the  data value are between 100000 and 1000000000。

**Comment by ConRresIm** [comment]
The error mainly occurs when the data value are between 100000 and 1000000000

**Comment by ConRresIm** [comment]
![image](https://user-images.githubusercontent.com/33385384/92191520-94f7e300-ee96-11ea-82f7-8c373926a7b3.png)
![Uploading image.png…]()

**Comment by jvmncs** [comment]
**TL;DR** With default settings, Pond is only correct for decimals up to 10^8 - 1. Outside of this bound, the protocol fails (but the security guarantee remains). This can be relaxed & configured using the `fixedpoint_config` parameter.

TFE (and specifically Pond) implements a fixed-point arithmetic to emulate the usual floating point ops common in TensorFlow. This is controlled by the `fixedpoint_config` parameter provided to the Pond protocol constructor, and its defaults are defined below:

https://github.com/tf-encrypted/tf-encrypted/blob/8b7cfb32c426e9a6f56769a1b47626bd1be03a66/tf_encrypted/protocol/pond/pond.py#L116-L130

On most platforms, the first condition will evaluate to True by default, so we will end up with a fixed point config of `fixed64`.
That is defined as follows:

https://github.com/tf-encrypted/tf-encrypted/blob/8b7cfb32c426e9a6f56769a1b47626bd1be03a66/tf_encrypted/tensor/fixed.py#L73-L80

The relevant number in your case is `precision_integral=7`, which says that this fixed-point config can only represent decimals less than 10^(7+1) - 1. In this case, the decimal will be accurate to 8 decimal places (`precision_fractional=8`). These numbers (as well as the others) have been chosen to be able to fit into the `tf.int64` data type while still maintaining correctness & security.

**Comment by jvmncs** [comment]
Closing this since it's not a bug, feel free to follow up with a comment if that's not the case.

**Comment by ConRresIm** [comment]
![image](https://user-images.githubusercontent.com/33385384/92223077-84fdf480-eed2-11ea-8d76-f92e48969d86.png)
![image](https://user-images.githubusercontent.com/33385384/92223113-90512000-eed2-11ea-8152-744123423f80.png)

**Comment by ConRresIm** [comment]
![image](https://user-images.githubusercontent.com/33385384/92223198-b37bcf80-eed2-11ea-8b3c-eb537970fb67.png)
![image](https://user-images.githubusercontent.com/33385384/92223212-b8d91a00-eed2-11ea-9b5f-67d5cc29a4ef.png)

**Comment by ConRresIm** [comment]
> Closing this since it's not a bug, feel free to follow up with a comment if that's not the case.



> Closing this since it's not a bug, feel free to follow up with a comment if that's not the case.

> there are wrong, for decimals less than or equal to 10 ^ 8-1
